File: myapplication/src/main/java/com/example/myapplication/retrofit/RetrofitInterface.java
package com.example.myapplication.retrofit;

import com.example.myapplication.bean.Galleryclass;
import com.example.myapplication.bean.ImageDetail;
import com.example.myapplication.bean.ImageList;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Query;

/**
 * Created by hero on 2017/5/21.
 */

public class RetrofitInterface {
    public interface GetImageTypeList {
        @GET("/tnfs/api/classify")
        Call<List<Galleryclass>> getImageTypeList();
    }

    public interface GetImageListByType {
        @POST("/tnfs/api/list")
        Call<ImageList> getImageList(@Query("id") int id, @Query("page") int page, @Query("rows") int rows);
    }

    public interface GetImageDetial {
        @POST("/tnfs/api/show")
        Call<ImageDetail> getImageDetial(@Query("id") long id);
    }
}

File: gankio/src/main/java/com/example/gankio/retrofit/RetrofitInterface.kt
package com.example.gankio.retrofit
import com.example.gankio.Const
import com.example.gankio.model.WelfareResult
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path

/**
 * Created by za-wanghe on 2017/6/16.
 */

interface RequestWelfareService {
    @GET(Const.WELFARE_URL)
    fun getWelfareList(@Path("pageSize") pageSize: Int, @Path("page") page: Int): Call<WelfareResult>
}
